mineral-ui/mineral-ui
# [Tabs]: New Component

### Description
* Prevent Button icons from shrinking with long content
* Accept more style input types for `createStyledComponent`
* Update DialogBody so that content overflows correctly
* Refactor OverflowContainerWithShadows to place shadows _over_ the overflowed content and apply in x & y axes
* Remove style reset from Popover root
* Add Truncate component to programmatically wrap Tooltip around truncated content
* Add Tabs component
* Add Tabs website examples

### Motivation and context
Closes #146 

### Screenshots, videos, or demo, if appropriate

https://146-tabs--mineral-ui.netlify.com/components/tabs

### How to test

1. `cd mineral-ui && npm start`
1. Navigate to [Tabs](https://146-tabs--mineral-ui.netlify.com/components/tabs) and [Tab](https://146-tabs--mineral-ui.netlify.com/components/tab)
    1. Confirm all looks and works well
1. Navigate to DialogBody > [Scrolling](https://146-tabs--mineral-ui.netlify.com/components/dialog-body/scrolling)
    1. Confirm that overflow and box shadows work in all browsers
1. Navigate to [Dropdown](https://146-tabs--mineral-ui.netlify.com/components/dropdown), [Select](https://146-tabs--mineral-ui.netlify.com/components/select) or any other [Popover](https://146-tabs--mineral-ui.netlify.com/components/popover)-based component
    1. Ensure that removing the style reset didn't change styles
1. Navigate to Button > [Button with Icons](https://146-tabs--mineral-ui.netlify.com/components/button/icons)
    1. Confirm that adding more content does not affect Icon size

### Types of changes
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

### Checklist




### How does this PR make you feel?

![Homer Simpson clicking tab](https://media.giphy.com/media/aYzxVt2lMrZXW/giphy.gif)
